src/main/java/hudson/remoting/PingThread.java

```diff
@@ -61,14 +61,24 @@
      */
     private final long interval;
 
-    public PingThread(Channel channel, long timeout, long interval) {
+    /**
+     * Specifis if troubleshooting information should be catched in case of pingFailure
+     */
+    private final boolean pingFailureAnalyzer;
+
+    public PingThread(Channel channel, long timeout, long interval, boolean pingFailureAnalyzer) {
```

Comment:
I would prefer to have it disabled in order ensure any bad effect of this PR on the remoting behaviour. In any case, I am happy to do whatever you think might be the best option here.